### ISSUE TITLE
Lambda ASP.NET Core Web API - Container Image does not work with .NET 7

### ISSUE THREAD

**Issue by IvanFarkas** [aws/aws-lambda-dotnet#1416]
### Describe the bug

# Lambda ASP.NET Core Web API - Container Image does not work with .NET 7

# Error in log:
```
RequestId: b5d40869-04bc-452b-8d82-ffa4134d670c Error: fork/exec /lambda-entrypoint.sh: exec format error
Runtime.InvalidEntrypoint
```

The same project configured for .NET 6 works, but configured for .NET 7 does not work.
See configuration differences in [Readme](https://github.com/IvanFarkas/serverless_image_AspNetCoreWebAPI/blob/net7/Readme.md)

**[Sample Project (.NET 6 - main branch)](https://github.com/IvanFarkas/serverless_image_AspNetCoreWebAPI)**
**[Sample Project (.NET 7 - net7 branch)](https://github.com/IvanFarkas/serverless_image_AspNetCoreWebAPI/tree/net7)**

### global.json

```json
"sdk": { "version": "6.0.405" }
```

### *.csproj 

```xml
<TargetFramework>net6.0</TargetFramework>
```

### aws-lambda-tools-defaults.json

```xml
"framework": "net6.0",
"function-runtime": "dotnet6",
"function-architecture": "arm64",
```

### Dockerfile

```Dockerfile
FROM public.ecr.aws/lambda/dotnet:6
```

### [AWS SAM Lambda Output](https://0s785ktk0i.execute-api.us-east-1.amazonaws.com)

```
AWS SAM WebAPI (.NET 6.0.10, amzn.2-arm64, Arm64, Arm64) on AWS Lambda with SAM

processor	: 0
BogoMIPS	: 243.75
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x3
CPU part	: 0xd0c
CPU revision	: 1

processor	: 1
BogoMIPS	: 243.75
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x3
CPU part	: 0xd0c
CPU revision	: 1
```

### global.json

```json
"sdk": { "version": "7.0.102" }
```

### *.csproj 

```xml
<TargetFramework>net7.0</TargetFramework>
```

### aws-lambda-tools-defaults.json

```xml
"framework": "net7.0",
"function-runtime": "dotnet7",
"function-architecture": "arm64",
```

### Dockerfile

```Dockerfile
FROM public.ecr.aws/lambda/dotnet:7
```

### [AWS SAM Lambda Output](https://0s785ktk0i.execute-api.us-east-1.amazonaws.com)

```json
{ "message": "Internal Server Error" }
```

### AWS Log

```
RequestId: b5d40869-04bc-452b-8d82-ffa4134d670c Error: fork/exec /lambda-entrypoint.sh: exec format error
Runtime.InvalidEntrypoint
```

### Expected Behavior

It behaves as the .NET 6 configuration

### Current Behavior

Report error. I suspect it does not work with amd64 Graviton CPU.

### AWS SAM Lambda Output

```json
{ "message": "Internal Server Error" }
```

### AWS Log

```
RequestId: b5d40869-04bc-452b-8d82-ffa4134d670c Error: fork/exec /lambda-entrypoint.sh: exec format error
Runtime.InvalidEntrypoint

### Reproduction Steps

- Clone [serverless_image_AspNetCoreWebAPI net7 branch](https://github.com/IvanFarkas/serverless_image_AspNetCoreWebAPI/tree/net7)
- Right-mouse click on the project and select **Publish to AWS Lambda...**
- Click endpoint URL
- Witness error

### Possible Solution

_No response_

### Additional Information/Context

Function ARN: arn:aws:lambda:us-east-1:981910913386:function:serverless-image-aspnetcoreweba-AspNetCoreFunction-zg2kwBNOkjMA
Request ID: ec2224f7-8ea1-4431-80f2-a5f7ab1df2c1
Timestamp with timezone:  2023-01-21T01:08:11.647Z

This is reported this as well at AWS Support, [Case ID 11809880271](https://support.console.aws.amazon.com/support/home?region=us-east-1#/case/?displayId=11809880271&language=en).

### AWS .NET SDK and/or Package version used

.NET SDK 7.0.102
Amazon.Lambda.AspNetCoreServer 7.3.0

### Targeted .NET Platform

.NET 7 (latest)

### Operating System and version

AmazonLinux

**Comment by rajeshaz09** [comment]
I read earlier that .....net7 only available in AMD64. 
Only LTS versions supported in ARM64 architecture .
I am not sure this changed now.

**Comment by ashishdhingra** [comment]
> Sample Project (.NET 7 - net7 branch)

@IvanFarkas Could you please confirm if changing architecture to `x86_64` works?

**Comment by IvanFarkas** [comment]
@ashishdhingra  Hi Ashish,
I confirm that it does work with the `x86_64` architecture.
I updated my [repo](https://github.com/IvanFarkas/serverless_image_AspNetCoreWebAPI/tree/net7) with the configuration and output.

What's the ETA of the `arm64` (Graviton) architecture for .NET 7 Lambda Container Image, please?

**Comment by IvanFarkas** [comment]
The next question is, which configuration should I use till `arm64` .NET 7 is OK?

**Factors**
- Startup time
- Run time
- Price

**Comment by normj** [comment]
@IvanFarkas We are blocked on ARM64 support till an upcoming change by Microsoft to .NET is released. The good news is there is good progress on Microsoft's side to get the change out. https://github.com/dotnet/runtime/issues/69361#issuecomment-1377850973

**Comment by ashishdhingra** [comment]
Duplicate of https://github.com/aws/aws-lambda-dotnet/issues/1310.

**Comment by IvanFarkas** [comment]
@normj Thx Norm.
